Problem after upgrading from Astro 5 to 6: Astro Actions fail when `output: 'static'` (Netlify adapter)

### Astro Info

```block
Astro                    v6.1.0
Vite                     v7.3.1
Node                     v24.14.0
System                   Linux (x64)
Package Manager          npm
Output                   static
Adapter                  @astrojs/netlify (v7.0.5)
Integrations             none
```

### If this issue only occurs in one browser, which browser is a problem?

All

### Describe the Bug

Since upgrading to Astro 6, Astro Actions no longer work with @astrojs/netlify adapter when `output` is set to `'static'`. 

The **_build and dev server_** fail with:


[ERROR] [astro:actions] An unhandled error occurred while running the "astro:routes:resolved" hook
[ActionsWithoutServerOutputError] A server is required to create callable backend functions. To deploy routes to a server, add an adapter to your Astro config and configure your route for on-demand rendering

Changing `output` to `'server'` resolves the issue — however, this was **not required in Astro 5**, where Actions worked fine with the default static output mode and the Netlify adapter.

This appears to be a breaking change / regression.

I had a similar issue with the netlify adapter: when using server:defer there was a problem on the netlify server with server islands (#16049 ). 
The fix was a simple change in order of execution by the netlify adapter. I suspect a similar issue here.


### What's the expected result?

The output "static" config should suffice to use Astro Actions with the netlify adapter. 

### Link to Minimal Reproducible Example

https://github.com/PanRoR/astro-server-island-netlify

### Participation

- [ ] I am willing to submit a pull request for this issue.

--- a/.changeset/actions-static-with-adapter.md
+++ b/.changeset/actions-static-with-adapter.md
@@ -1,0 +1,5 @@
+---
+'astro': patch
+---
+
+Fixes Actions failing with `ActionsWithoutServerOutputError` when using `output: 'static'` with an adapter

--- a/packages/astro/src/actions/integration.ts
+++ b/packages/astro/src/actions/integration.ts
@@ -41,7 +41,7 @@ export default function astroIntegrationActionsRouteHandler({
 				});
 			},
 			'astro:routes:resolved': ({ routes }) => {
-				if (!hasNonPrerenderedRoute(routes)) {
+				if (!settings.config.adapter && !hasNonPrerenderedRoute(routes)) {
 					const error = new AstroError(ActionsWithoutServerOutputError);
 					error.stack = undefined;
 					throw error;

--- a/packages/astro/test/actions.test.js
+++ b/packages/astro/test/actions.test.js
@@ -476,6 +476,25 @@ describe('Astro Actions in static mode with prerender = false routes', () => {
 	});
 });
 
+it('Works with adapter and all pages prerendered', async () => {
+	const fixture = await loadFixture({
+		root: './fixtures/actions/',
+		output: 'static',
+		adapter: testAdapter(),
+	});
+	const devServer = await fixture.startDevServer();
+	const res = await fixture.fetch('/_actions/subscribe', {
+		method: 'POST',
+		headers: {
+			'Content-Type': 'application/json',
+		},
+		body: JSON.stringify({ channel: 'bholmesdev' }),
+	});
+
+	assert.equal(res.ok, true);
+	await devServer.stop();
+});
+
 it('Base path should be used', async () => {
 	const fixture = await loadFixture({
 		root: './fixtures/actions/',

